circus/plugins/watchdog.py

```diff
@@ -217,6 +223,10 @@ def look_after(self):
 
         max_timeout = self.loop_rate * self.max_count
         too_old_time = time.time() - max_timeout
+        """Instead of del in loop, that will cause exception in Python3,
+        add to list and del from pid_status after loop.
+        """
```

Comment:
Another solution would be to make a copy of the dict before the `for`, but your solutions is fine too :) .

Could you make your comment as a real comment (`# ...` and not `""" ... """`) though ?